### PR TITLE
NO-JIRA: Add Network Layer

### PR DESCRIPTION
**Pull request by igorTimofiejczyk**

### CODE REVIEW

File: animeal/src/Networking/NetworkService.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import AWSDataStorePlugin
+import AWSAPIPlugin
+import AWSPluginsCore
+import UIComponents
+
+protocol NetworkServiceHolder {
+    var networkService: NetworkServiceProtocol { get }
+}
+
+typealias MyClientResponse<ResponseType: Decodable> = Result<ResponseType, Error>
```
Comment (diantyn) [inline animeal/src/Networking/NetworkService.swift]: Maybe we can use Result as is, without type alias ?

File: animeal/src/Networking/NetworkRequest.swift [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import AWSPluginsCore
+import Amplify
+
+struct Request<R: Decodable> {
```
Comment (diantyn) [inline animeal/src/Networking/NetworkRequest.swift]: I suggest moving that generic request and service to services package, mapping will be leave here

File: animeal/src/Networking/NetworkRequest.swift [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    let apiName: String?
+    let document: String
+    let variables: [String: Any]?
+    let responseType: R.Type
+    let decodePath: String?
+
+    init(
+        apiName: String? = nil,
```
Comment (diantyn) [inline animeal/src/Networking/NetworkRequest.swift]: nit: Maybe we can also write parameters without default values first and then parameters with default values?

File: animeal/src/Networking/NetworkRequest.swift (hunk 2 above)
Comment (igorTimofiejczyk) [reply animeal/src/Networking/NetworkRequest.swift]: Done

File: animeal/src/Networking/NetworkRequest.swift (hunk 3 above)
Comment (igorTimofiejczyk) [reply animeal/src/Networking/NetworkRequest.swift]: Done

File: animeal/src/Networking/NetworkService.swift (hunk 1 above)
Comment (igorTimofiejczyk) [reply animeal/src/Networking/NetworkService.swift]: Done

**Review by diantyn — APPROVED** [review]

**Pull request merged**